problems/arraysWithLoops.js

```diff
@@ -25,39 +38,62 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+function shoutForOfLoop(array) {
+  let array2 = []
+  for(let element of array) array2.push(element + "!");
+  return array2;
+}
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+function sumArray(nums) {
+  let sum = 0
+  for(let i = 0; i <= nums.length - 1; i++) sum += nums[i];
+  return sum;
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+function oddArray(nums) {
+  let sum = []
+  for(let i = 0; i <= nums.length - 1; i++)
+  if (nums[i] % 2 === 1) sum.push(nums[i]);
+  return sum;
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
+function evenArray(nums) {
+  let sum = []
+  for(let i = 0; i <= nums.length - 1; i++)
+  if(nums[i] % 2 === 0) sum.push(nums[i]);
```

Comment:
There should be curly braces after the if conditional and the return statement should be tabbed within.